spec/policies/liquid_tag_policy_spec.rb

```diff
@@ -0,0 +1,64 @@
+require "rails_helper"
+
+RSpec.describe LiquidTagPolicy, type: :policy do
+  let(:liquid_tag) { instance_double(Liquid::Tag) }
+
+  describe "initialize?" do
+    let(:action) { :initialize? }
+    let(:article) { create(:article) }
+
+    it "raises an error if user is missing" do
+      user = nil
+      parse_context = { source: article, user: user }
+      allow(liquid_tag).to receive(:parse_context).and_return(parse_context)
+      stub_const("#{liquid_tag.class}::VALID_ROLES", [:admin])
+      expect do
+        Pundit.authorize(user, liquid_tag, action, policy_class: described_class)
+      end.to raise_error(Pundit::NotAuthorizedError, "No user found")
+    end
+
+    it "authorizes and skips logic if liquid tag is not role restricted" do
+      user = create(:user)
+      parse_context = { source: article, user: user }
+      allow(liquid_tag).to receive(:parse_context).and_return(parse_context)
+      allow(user).to receive(:has_role?)
+      expect do
+        Pundit.authorize(user, liquid_tag, action, policy_class: described_class)
+      end.not_to raise_error
+      expect(user).not_to have_received(:has_role?)
+    end
+
+    it "authorizes if user has the correct role" do
+      user = create(:user, :admin)
+      parse_context = { source: article, user: user }
+
+      allow(liquid_tag).to receive(:parse_context).and_return(parse_context)
+      stub_const("#{liquid_tag.class}::VALID_ROLES", [:admin])
+      expect do
+        Pundit.authorize(user, liquid_tag, action, policy_class: described_class)
+      end.not_to raise_error
+    end
+
+    it "handles single resource roles" do
+      # TODO: (Alex Smith) - update roles to new liquid tag role for more relevant example/use
+      user = create(:user, :single_resource_admin, resource: Article)
+      parse_context = { source: article, user: user }
+
+      allow(liquid_tag).to receive(:parse_context).and_return(parse_context)
+      stub_const("#{liquid_tag.class}::VALID_ROLES", [[:single_resource_admin, Article]])
```

Comment:
Hm, I am confused about this test. Looking [at the docs](https://github.com/thepracticaldev/dev.to/pull/8917/files#diff-050073bb34d810dba4e9692f8aca0209R128), shouldn't the resource here be `NewLiquidTag` rather than `Article`? Or am I misunderstanding?

Reply:
`NewLiquidTag` isn't defined, so we can't use that in code. That was included in the docs to demonstrate how it would be used. Whatever new tag you're creating, you could include that as a role.